### DISCUSSION TITLE
Components with SVGs

### DISCUSSION THREAD

**Discussion by MartinHughes-BPC** [tailwindlabs/tailwindcss#3038] [Help] [answered]
Hi All,

I'm making a Blade nav-item component (basically one of the TailwindUI examples.So far I have this (I've redacted most of the CSS to avoid stamping all over Adam's copyright):

```
<a href="{{ $url }}" class="
    {{ url()->full() == $url
        ? 'bg-indigo-800...'
        : 'text-indigo-100...'}}">

    <svg class="text-indigo-300 mr-3 h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6" />
    </svg>
    {{ $name }}
</a>
```
I'm wondering what the best/most sensible approach to handle the SVGs is. I can obviously use a slot to inject the SVGs, however I want to avoid repeating the CSS. Would people generally recommend turning each SVG icon into its own component? Should I simply make the svg path element the slot and inject that instead?

Any advice greatly appreciated!

Thanks,
Martin.

**Comment by ihorvorotnov** [comment, marked as answer]
Check this package for icons in Blade: https://github.com/blade-ui-kit/blade-icons

**Reply by MartinHughes-BPC** [reply]
Cool. Thanks for that!
